server/src/LspServer.ts

```diff
@@ -414,6 +437,24 @@ export class LspServer {
     }
   }
 
+  onDidCreateFiles(_params: CreateFilesParams): void {
+    this.dbt?.refresh();
+  }
+
+  onDidRenameFiles(params: RenameFilesParams): void {
```

Comment:
After rename LS receives document opened event for new file. But the old one is still in memory of LS. It seems that we need to do anything with old file. Let's try:
1. Open some dbt model
2. Make dbt error
3. Open another dbt model
4. Rename first file
5. Fix dbt error in first file
6. Global error will not be cleared

Moreover if you will create new file with name equal to previous name of the first it will show old compile result.

Reply:
Thanks, I added fix for this case, please take a look.